addons/marketing_card/models/card_campaign.py

```diff
@@ -262,23 +350,21 @@ def _compute_render_model(self):
         for campaign in self:
             campaign.render_model = campaign.res_model
 
-    def _get_card_element_values(self, record, preview_values):
-        """Helper to get the right value for each element when rendering."""
+    def _get_card_element_values(self, record):
+        """Helper to get the right value for dynamic fields."""
         self.ensure_one()
-        value_from_role = {}
-        default_values = {
-            'background': self.card_template_id.default_background
+        result = {
+            'image1': images[0] if (images := self.content_image1_path and record.mapped(self.content_image1_path)) else False,
+            'image2': images[0] if (images := self.content_image2_path and record.mapped(self.content_image2_path)) else False,
         }
-        for element in self.card_element_ids:
-            value = element._get_render_value(record)
-            if not value and element.card_element_role in default_values:
-                value = default_values[element.card_element_role]
-            if not value and preview_values and element.card_element_role in preview_values and element.value_type == 'static':
-                value = preview_values[element.card_element_role]
-            if not value and not record:
-                value = element._get_placeholder_value()
-            value_from_role[element.card_element_role] = value
-
-        # in qweb t-out of "False" effectively removed the element while '' does not
-        # we force everything to '' to be consistent
-        return {element: val or '' for element, val in value_from_role.items()}
+        for el in ('header', 'sub_header', 'section', 'sub_section1', 'sub_section2'):
```

Comment:
Personal taste: would better have a clear mapping header -> content_header_dyn, content_header_path, ...  that way when you grep fields (something I regularly do in order to see a bit how they are used n code), you find where they are used (otherwise you will never find this code when grepping)